starknet/l2_minter/src/helpers/booking_storage.cairo

```diff
@@ -0,0 +1,93 @@
+        StorageAccessBooking::read_at_offset_internal(address_domain, base, 0_u8)
+    }
+    fn read_at_offset_internal(
+        address_domain: u32, base: StorageBaseAddress, offset: u8
+    ) -> SyscallResult::<Booking> {
+        Result::Ok(
+            Booking {
+                value: u256 {
+                    low: storage_read_syscall(
+                        address_domain, storage_address_from_base_and_offset(base, 0_u8 + offset)
+                    )?
+                        .try_into()
+                        .unwrap(),
+                    high: storage_read_syscall(
+                        address_domain, storage_address_from_base_and_offset(base, 1_u8 + offset)
+                    )?
+                        .try_into()
+                        .unwrap()
+                    }, amount: u256 {
+                    low: storage_read_syscall(
+                        address_domain, storage_address_from_base_and_offset(base, 2_u8 + offset)
+                    )?
+                        .try_into()
+                        .unwrap(),
+                    high: storage_read_syscall(
+                        address_domain, storage_address_from_base_and_offset(base, 3_u8 + offset)
+                    )?
+                        .try_into()
+                        .unwrap()
+                },
+                status: storage_read_syscall(
+                    address_domain, storage_address_from_base_and_offset(base, 4_u8 + offset)
+                )?
+                    .try_into()
+                    .unwrap(),
+            }
+        )
+    }
+
+    fn write(address_domain: u32, base: StorageBaseAddress, value: Booking) -> SyscallResult::<()> {
+        StorageAccessBooking::write_at_offset_internal(address_domain, base, 0_u8, value)
+    }
+
+    fn write_at_offset_internal(
+        address_domain: u32, base: StorageBaseAddress, offset: u8, value: Booking
+    ) -> SyscallResult::<()> {
+        storage_write_syscall(
+            address_domain,
+            storage_address_from_base_and_offset(base, 0_u8 + offset),
+            value.value.low.into()
+        )?;
+        storage_write_syscall(
+            address_domain,
+            storage_address_from_base_and_offset(base, 1_u8 + offset),
+            value.value.high.into()
+        )?;
+        storage_write_syscall(
+            address_domain,
+            storage_address_from_base_and_offset(base, 2_u8 + offset),
+            value.amount.low.into()
+        )?;
+        storage_write_syscall(
+            address_domain,
+            storage_address_from_base_and_offset(base, 3_u8 + offset),
+            value.amount.low.into()
+        )?;
+        storage_write_syscall(
+            address_domain,
+            storage_address_from_base_and_offset(base, 4_u8 + offset),
+            value.status.into()
+        )
+    }
+
+    fn size_internal(value: Booking) -> u8 {
+        5_u8
+    }
+}
```

Comment:
Is all this impl required to store a struct?
Because the corresponding package import is commented in the minter contract, I understand that it is not more required?

Reply:
It isn't required anymore thanks to the new StorageAccess derivation but this could change with Enums in the future. I decided to keep it just in case.